lstchain/scripts/lstchain_dl1_to_dl2.py

```diff
@@ -67,7 +67,9 @@ def main():
     data = pd.read_hdf(args.datafile, key=dl1_params_lstcam_key)
 
     if config['source_dependent']:
-        data = pd.concat([data, pd.read_hdf(data, key=dl1_params_src_dep_lstcam_key)], axis=1)
+        data = pd.concat([data, pd.read_hdf(args.datafile, key=dl1_params_src_dep_lstcam_key)], axis=1)
+        data = data.drop("index", axis=1)
```

Comment:
Why do you need to drop "index" here?

I think if there is a column "index" in the file, that is not recognized as **the** index when reading it with pandas, either the thing writing it should make something different.

Reply:
After adding the source parameters using `lstchain_add_source_dependent_parameters.py`, `index` column is apparently included in `dl1_params_src_dep_lstcam_key`.
Actually `write_dataframe` saved this `index` column.

<img width="522" alt="dl1_src_dep" src="https://user-images.githubusercontent.com/31307823/80130504-a7523e00-85d3-11ea-8c25-74fa6b6c5a68.PNG">

Maybe due to this `index` column, I got the following error  during ` write_dl2_dataframe` in `lstchain_dl1_to_dl2.py`
```
ValueError: appended items do not match existing items in table!
```

Do you have any suggestions...?

Reply:
Most pandas write methods have an `index` keyword argument, that defines if the index should be stored to file. 

Normally you want that, and normally pandas detects that again on reading the file and automatically assigns that column as the `index` of the `DataFrame` you get. 

Somehow, it seems to be duplicated here (observe how the column `index` is identical to the actual index of the DataFrame.

A question would be, if the DataFrame you pass to `write_dataframe` already has the `index` column or if the `write_dataframe` function calls `reset_index`.

Reply:
In any case, that column should not be in the file as a column.

Reply:
I fixed it by using `set_index` when reading.